mlir/lib/Quantum/Transforms/ConversionPatterns.cpp

```diff
@@ -415,8 +415,30 @@ struct InsertOpPattern : public OpConversionPattern<InsertOp> {
     LogicalResult matchAndRewrite(InsertOp op, InsertOpAdaptor adaptor,
                                   ConversionPatternRewriter &rewriter) const override
     {
-        // Unravel use-def chain of quantum register values, converting back to reference semantics.
-        rewriter.replaceOp(op, adaptor.getInQreg());
```

Comment:
I might be nice to gate this new behaviour behind some flag? Not a must, but since you mention several pitfalls that  could allow us to test it out safely